tests/integrationv2/test_buffered_send.py

```diff
@@ -0,0 +1,119 @@
+import pytest
+
+from configuration import available_ports, PROTOCOLS, Protocols, ALL_TEST_CIPHERS, Certificates
+from common import ProviderOptions, data_bytes
+from fixtures import managed_process # lgtm [py/unused-import]
+from providers import Provider, S2N, OpenSSL, GnuTLS
+from utils import invalid_test_parameters, get_parameter_name, to_bytes
+
+SEND_DATA_SIZE = 2 ** 16
+
+K_BYTES = 1024 
+SEND_BUFFER_SIZE_MIN = 1031
+SEND_BUFFER_SIZE_MIN_RECOMMENDED = 2 * K_BYTES
+SEND_BUFFER_SIZE_MULTI_RECORD = 17 * K_BYTES
+SEND_BUFFER_SIZE_PREFER_THROUGHPUT = 35 * K_BYTES
+SEND_BUFFER_SIZE_HUGE = 512 * K_BYTES
+
+SEND_BUFFER_SIZES = [
+    SEND_BUFFER_SIZE_MIN,
+    SEND_BUFFER_SIZE_MIN_RECOMMENDED,
+    SEND_BUFFER_SIZE_MULTI_RECORD,
+    SEND_BUFFER_SIZE_PREFER_THROUGHPUT,
+    SEND_BUFFER_SIZE_HUGE
+]
+
+TEST_CERTS = [
+    Certificates.RSA_4096_SHA512,
+    Certificates.ECDSA_384,
+    Certificates.RSA_PSS_2048_SHA256
+]
+
+@pytest.mark.uncollect_if(func=invalid_test_parameters)
+@pytest.mark.parametrize("cipher", ALL_TEST_CIPHERS, ids=get_parameter_name)
+@pytest.mark.parametrize("other_provider", [S2N, OpenSSL, GnuTLS], ids=get_parameter_name)
+@pytest.mark.parametrize("provider", [S2N], ids=get_parameter_name)
+@pytest.mark.parametrize("protocol", PROTOCOLS, ids=get_parameter_name)
+@pytest.mark.parametrize("certificate", TEST_CERTS, ids=get_parameter_name)
+@pytest.mark.parametrize("buffer_size", SEND_BUFFER_SIZES, ids=get_parameter_name)
+@pytest.mark.parametrize("fragment_preference", [None, "--prefer-low-latency", "--prefer-throughput"], ids=get_parameter_name)
+def test_s2n_buffered_send_server(managed_process, cipher, other_provider, provider, protocol, certificate, buffer_size, fragment_preference):
+    port = next(available_ports)
+    random_bytes = data_bytes(SEND_DATA_SIZE)
+
+    client_options = ProviderOptions(
+        mode=Provider.ClientMode,
+        port=port,
+        cipher=cipher,
+        data_to_send=None,
+        insecure=True,
+        protocol=protocol)
+
+    extra_flags = ['--buffered-send', buffer_size] + \
+        ([] if fragment_preference is None else [fragment_preference])
```

Comment:
The thought is that by eliminating control flow there is less to think about.  In the implementation I have there is just one expression to think about.

I also thought about 

```python
extra_flags = list(filter(lambda x: x is not None, ['--buffered-send', buffer_size, fragment_preference]))
```

But this seemed much simpler.

Reply:
But isn't there just as much control flow in the current code? You still have an if, it's just now part of some more complicated array math.

Your second option seems simpler, but also kind of deceptive since it's really only trying to filter out one None. But hey, I'm not a python expert :)